### ISSUE TITLE
Enable to set filter_already_liked_items in precision_at_k

### ISSUE THREAD

**Issue by joddm** [benfred/implicit#171]
Hi, 

Would it make sense to be able to set the argument for `filter_already_liked_items` to False in the `precision_at_k` evaluation method?

Cheers

**Comment by medoidai** [comment]
@paulvercoustre 

Hi!

Could you please explain me: Why is not correct to use filter_already_liked_items=True when evaluating metrics on test data?

Currently, the ranking_metrics_at_k uses model.recommend() with default value True in filter_already_liked_items which I think is correct so that the metric will be measured only in the provided hold out set. Otherwise, you will measure not only in the test data but also train.

The logic internally is the following: For all the users in the test set that have at least one interaction we get recommendations for him/her except for the items found in the training dataset. Later on we compare the recommendation list from the model with the items the user has interacted with in the test set.

**Comment by paulvercoustre** [comment]
@medoidai 

I think you misunderstood my comment, it might not have been clear.

I agree with what you said. Setting filter_already_liked_items to True is the fair way to evaluate your model.

**Comment by medoidai** [comment]
Perfect, thank you!
